### DISCUSSION TITLE
How to add key-value pair to an existing file at a specific location?

### DISCUSSION THREAD

**Discussion by SpatikaIyengar** [nlohmann/json#2636] [Q&A] [answered]
I have a json file

```json
{
"key1":"value1",
"key2":2,
"object":{ "obj1":"v1",
"obj2":"v2"
}
}
```

I would like to add an entry {"z":5} to the existing json file at a specific location, say after "key2". Could anyone please suggest how to do that? I tried push_back but the file is written this way:

```json
{
"key1":"value1",
"key2":2,
"object":{ "obj1":"v1",
"obj2":"v2"
}
}
{
"z":5
}
```

When I try emplace the entry is added at the end. I believe it automatically sorts the keys.

Thank you very much for your help!
Spatika

**Comment by nlohmann** [comment, marked as answer]
JSON objects have no order, so you cannot add an entry *after* a key. See https://github.com/nlohmann/json#order-of-object-keys for more information.

**Reply by nlohmann** [reply]
@SpatikaIyengar Do you need further assistance?

**Comment by SpatikaIyengar** [comment]
My apologies for not replying earlier. I don't need further assistance.
Thank you very much for your time and help!
-Spatika
---


*Dr. Spatika Iyengar*
Senior CFD Modeler,
*AquaNRG Consulting Inc.,*
a: 405 Main Street, Suite # 807
Houston, TX 77002
Tel: +1(386)-333-4259


*http://www.aquanrgconsulting.com/ <http://www.aquanrgconsulting.com/>*

*https://twitter.com/AquaNRG <https://twitter.com/AquaNRG>*
*https://www.linkedin.com/in/spatikaiyengar
<https://www.linkedin.com/in/spatikaiyengar>*
